kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/Referent.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.resolver;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressStatus;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyStatus;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaServiceStatus;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaClusterStatus;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilterStatus;
+import io.kroxylicious.kubernetes.operator.ResourcesUtil;
+
+/**
+ * A single referent
```

Comment:
* This is really modelling the state of all the references from the referrer, I think? If so, perhaps there's a better name for this type.
* Could you give an example? I think that would make it much more concrete for the reader.

Reply:
Referent is a little wrapper class around the Referent resource so there is a common interface for finding stale or ResolvedRefs=False Referents. Ideally we'd be able to have these interfaces directly on the Resource classes.

Maybe calling it a `Referent` isn't that useful, it's describing something about what status/conditions the resource has? HasCommonStatus? KroxyliciousResource? Something saying it has some features common to all our CRs.

Reply:
Have removed `Referent`, instead the user can get at the resource directly and ask if it's stale/hasResolvedRefsFalseCondition via static methods on ResourcesUtil